### PR TITLE
feat(Page): add onScroll callbacks

### PR DESCRIPTION
**Pull request by hitpopdimestop**
This PR adds scroll listeners to Page component and resolves this feature request:
[#5596 ](https://github.com/wix/wix-style-react/issues/5596)

### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [x] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5669/

File: src/Page/Page.js [hunk 1]
```diff
@@ -95,6 +99,9 @@ class Page extends React.PureComponent {
     // Maybe there is a transition
     const ARBITRARY_SHORT_DURATION_MS = 100;
     setTimeout(this._calculateComponentsHeights, ARBITRARY_SHORT_DURATION_MS);
+
+    this.props.scrollableContentRef &&
+      this.props.scrollableContentRef(this.scrollableContainerRef.current);
```
Comment (lironch) [inline src/Page/Page.js:104]: Consider adding a comment above these lines to make it clear it is not part of the "hack" above.
Another option is to restore the `_setScrollContainer` method, rename it to `_publishScrollableContainer` and call it here.

**Comment by lironch** [comment]
@shlomitc
Regarding the issue you pointed out with delaying the header from minimising according to the `scrollThrottleWait` value.
We need to remember that the situation before was much worst, we were handling the scroll event in an `active` way and blocking by that the rendering thread, now at least we provide a `passive` handling to this event using the throttle mechanism which should provide a much smoother experience to the user as long as the `scrollThrottleWait` is not a big value.

Here are the options i think we have moving forward:
1. Declare an internal value for `scrollThrottleWait` in the `Page` component without exposing it as a prop to the users for modifications. We can make sure the value is small enough to provide us with `passive` handling of the event while visually there is no noticeable change to the component behaviour. 
In terms of cost-effectiveness, this one is very "cheap" and should suffice for the time being which makes it my favorite option, if we will get requests for configurable `scrollThrottleWait` value in the future we will handle it then.
2. Pass a `ref` from `Page` to `ScrollableContainer` and register the `Page` component to the `scroll` events on that `ref`.
This means that `Page` will have full control on the event handling for it's internal usages while the event handling for the `Page` users is done by `ScrollableContainer` according to the provided `scrollThrottleWait` value they set. 
Cost-effectiveness wise, this one is more expensive since we will need to write and test more code in the `Page` component   which is already written and tested at `ScrollableContainer` so i this this can be done when option 1 won't be enough.

Regarding adding visual/e2e tests, i looked on the current suite we have and it felt "good enough" to cover this new addition, every new test i thought off felt to me like something that tests `ScrollableContainer` and not the integration, any ideas for uncovered flows?

**Comment by shlomitc** [comment]
Regarding the solution - if it doesn't need to be configurable, just remove the prop and leave some comment to explain that this value is hardcoded because it affects the page header animation.

I must say I don't really understand the difference between `onScrollChanged` and `onScrollPositionChanged`. Since there are no tests or examples, it's really hard to understand how to use the. the comments explain what they do, but not when or how should I use it. This is why I think tests can be important here

**Comment by lironch** [comment]
I created a [new PR](https://github.com/wix/wix-style-react/pull/5766) that addresses the issues in this one
FYI @hitpopdimestop @shlomitc

**Pull request closed without merging**